File: VB-Examples/10_Shapes/MiddleCenteredTextOfShape.vb
Imports Spire.Xls
Imports Spire.Xls.Core
Imports System.ComponentModel
Imports System.Text

Namespace MiddleCenteredTextOfShape
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object
            Dim workbook As New Workbook()

            ' Get the first worksheet from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Add a rectangle shape to the worksheet at position (8, 2) with size 300x300
            Dim rect As IPrstGeomShape = sheet.PrstGeomShapes.AddPrstGeomShape(8, 2, 300, 300, PrstGeomShapeType.Rect)

            ' Set the fill color of the rectangle shape to white
            rect.Fill.ForeColor = Color.White
            ' Set the fill type of the rectangle shape to solid color
            rect.Fill.FillType = ShapeFillType.SolidColor

            ' Set the text content of the rectangle shape to "E-iceblue"
            rect.Text = "E-iceblue"
            ' Set the vertical alignment of the text within the rectangle shape to centered
            rect.TextVerticalAlignment = ExcelVerticalAlignment.MiddleCentered

            ' Save the workbook to a file named "result.xlsx" in Excel 2013 format
            workbook.SaveToFile("result.xlsx", ExcelVersion.Version2013)
            ' Release the resources used by the workbook
            workbook.Dispose()

            'View the document
            FileViewer("result.xlsx")
		End Sub

		Private Sub FileViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub


	End Class
End Namespace

File: CS-Examples/10_Shapes/MiddleCenteredTextOfShape.cs
using Spire.Xls;
using Spire.Xls.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace MiddleCenteredTextOfShape
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void btnRun_Click(object sender, EventArgs e)
        {
            // Create a new workbook object
            Workbook workbook = new Workbook();

            // Get the first worksheet from the workbook
            Worksheet sheet = workbook.Worksheets[0];

            // Add a rectangle shape to the worksheet
            IPrstGeomShape rect = sheet.PrstGeomShapes.AddPrstGeomShape(8, 2, 300, 300, PrstGeomShapeType.Rect);

            // Set the fill color of the rectangle to white (solid color)
            rect.Fill.ForeColor = Color.White;
            rect.Fill.FillType = ShapeFillType.SolidColor;

            // Set the text content of the rectangle
            rect.Text = "E-iceblue";

            // Set the vertical alignment of the text to middle-centered
            rect.TextVerticalAlignment = ExcelVerticalAlignment.MiddleCentered;

            // Save the workbook to a file
            workbook.SaveToFile("result.xlsx", ExcelVersion.Version2013);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            //View the document
            FileViewer("result.xlsx");
        }

        private void FileViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }

       
    }
}
